libs/trainer/mlpe/trainer/optimizers.py

```diff
@@ -0,0 +1,34 @@
+import inspect
+from typing import Iterable
+
+import torch
+
+
+def _wrap_optimizer(optimizer):
+    def func(*args, **kwargs):
+        def f(parameters):
+            return optimizer(parameters, *args, **kwargs)
+
+        return f
+
+    params = inspect.signature(optimizer).parameters
+    params = list(params.values())[1:]
+    func.__signature__ = inspect.Signature(params)
+    return func
+
+
+# typeo requires type hints to parse arguments from the command line.
+# this is a thin wrapper that type hints torch.optim.Adam arguments.
+# if there is an argument you wish to specify from the command line,
+# you must add it here.
+class Adam(torch.optim.Adam):
+    def __init__(
+        self,
+        parameters: Iterable,
+        lr: float = 1e-3,
+        weight_decay: float = 0.0,
+    ):
+        super().__init__(parameters, lr=lr, weight_decay=weight_decay)
+
+
+optimizers = {"adam": _wrap_optimizer(Adam)}
```

Comment:
same comments at the wrappers around flow and embeddings. Add variable docstring, and describe usage, and future addition directives.